jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -422,24 +506,38 @@ object XGBoost extends Serializable {
       case "_reg_" =>
         val xgBoostModel = new XGBoostRegressionModel(SXGBoost.loadModel(dataInStream))
         setGeneralModelParams(featureCol, labelCol, predictionCol, xgBoostModel)
+      case other =>
+        throw new XGBoostError(s"Unknown model type $other. Supported types " +
+          s"are: ['_reg_', '_cls_'].")
     }
   }
 }
 
+/** Class to hold the training and test matrices in a given partition. */
 private class Watches private(val train: DMatrix, val test: DMatrix) {
-  def toMap: Map[String, DMatrix] = Map("train" -> train, "test" -> test)
-      .filter { case (_, matrix) => matrix.rowNum > 0 }
+
+  def toMap: Map[String, DMatrix] = {
+    Map("train" -> train, "test" -> test).filter { case (_, matrix) => matrix.rowNum > 0 }
+  }
```

Comment:
why this method needs {} but delete() not?

Reply:
Again, I'm happy to change it back. This method doesn't fit on one line, so generally, in my experience, curly braces are used to enclose the method body. `delete` fits on one line, which is the difference.

Reply:
Also, if there is a style guide please let me know for future reference :)

Reply:
we generally follow Spark's style which is regulated by scalastyle.xml, for the cases which are not covered by the plugin, we just keep consistent with the existing coding style.

In this case, when the method is doing something instead of simply returning a value (with very simple transformation), we encourage to wrap with `{}`.

Reply:
If I understand correctly, you agree with the change?

Reply:
no, toMap is simply returning the value, delete is bringing side effect